--- a/browser/base/content/browser-menubar.inc.orig       2019-10-22 00:25:42.885043896 +0000
+++ b/browser/base/content/browser-menubar.inc    2019-10-22 00:28:41.502321848 +0000
@@ -66,6 +66,10 @@
                 <menuitem id="goOfflineMenuitem"
                           type="checkbox"
                           command="cmd_toggleOfflineStatus" data-l10n-id="menu-file-go-offline"/>
+                <menuitem id="menu_restart"
+                          label="&menu_restart.label;"
+                          accesskey="&menu_restart.accesskey;"
+                          command="cmd_restart"/>              
 <!-- We need to leave those strings in DTD until bug 1568133 is fixed. -->
                 <menuitem id="menu_FileQuitItem"
 #ifdef XP_WIN
diff -r 84219fbf133c browser/base/content/browser-sets.inc
--- a/browser/base/content/browser-sets.inc     Wed Oct 17 18:50:12 2018 +0000
+++ b/browser/base/content/browser-sets.inc     Tue Oct 23 18:42:49 2018 -0700
@@ -37,6 +37,7 @@
     <command id="cmd_closeWindow" oncommand="BrowserTryToCloseWindow()"/>
     <command id="cmd_toggleMute" oncommand="gBrowser.selectedTab.toggleMuteAudio()"/>
     <command id="cmd_CustomizeToolbars" oncommand="gCustomizeMode.enter()"/>
+    <command id="cmd_restart" oncommand="restartApplication()"/>
     <command id="cmd_toggleOfflineStatus" oncommand="BrowserOffline.toggleOfflineStatus();"/>
     <command id="cmd_quitApplication" oncommand="goQuitApplication()"/>

diff -r 150c70d39363 browser/base/content/browser.js
--- a/browser/base/content/browser.js	Tue Sep 18 16:08:09 2018 +0000
+++ b/browser/base/content/browser.js	Wed Sep 26 03:08:16 2018 -0700
@@ -8020,6 +8020,18 @@
   }
 };
 
+// Restart the browser
+function restartApplication() {
+  let cancelQuit = Cc["@mozilla.org/supports-PRBool;1"].
+                  createInstance(Ci.nsISupportsPRBool);
+  Services.obs.notifyObservers(cancelQuit, "quit-application-requested", "restart")
+
+  if (cancelQuit.data)
+    return;
+
+  Services.startup.quit(Ci.nsIAppStartup.eRestart | Ci.nsIAppStartup.eAttemptQuit);
+}
+
 // Prompt user to restart the browser in safe mode
 function safeModeRestart() {
   if (Services.appinfo.inSafeMode) {
diff -r 150c70d39363 browser/locales/en-US/chrome/browser/browser.dtd
--- a/browser/locales/en-US/chrome/browser/browser.dtd	Tue Sep 18 16:08:09 2018 +0000
+++ b/browser/locales/en-US/chrome/browser/browser.dtd	Wed Sep 26 03:08:16 2018 -0700
@@ -692,6 +692,9 @@
 <!ENTITY sidebarCloseButton.tooltip     "Close sidebar">
 <!ENTITY sidebarMenuClose.label         "Close Sidebar">
 
+<!ENTITY menu_restart.label            "Restart">
+<!ENTITY menu_restart.accesskey                "R">
+
 <!ENTITY quitApplicationCmdWin2.label       "Exit">
 <!ENTITY quitApplicationCmdWin2.accesskey   "x">
 <!ENTITY quitApplicationCmdWin2.tooltip     "Exit &brandShorterName;">
