diff -r c2db4a50dc5c browser/extensions/activity-stream/lib/ActivityStream.jsm
--- a/browser/extensions/activity-stream/lib/ActivityStream.jsm	Wed Jan 24 14:56:48 2018 -0800
+++ b/browser/extensions/activity-stream/lib/ActivityStream.jsm	Wed Jan 31 23:16:11 2018 -0800
@@ -190,7 +190,7 @@
         "CA": ["en-US", "en-GB", "en-ZA"],
         "DE": ["de", "de-DE", "de-AT", "de-CH"]
       })[geo];
-      return !!locales && locales.includes(locale);
+      return false;
     }
   },
   {
