diff -r 150c70d39363 browser/extensions/activity-stream/lib/ActivityStream.jsm
--- a/browser/extensions/activity-stream/lib/ActivityStream.jsm	Tue Sep 18 16:08:09 2018 +0000
+++ b/browser/extensions/activity-stream/lib/ActivityStream.jsm	Wed Sep 26 03:09:17 2018 -0700
@@ -46,7 +46,7 @@
 
 // Determine if spocs should be shown for a geo/locale
 function showSpocs({geo}) {
-  return SPOCS_GEOS.includes(geo);
+  return false;
 }
 
 // Configure default Activity Stream prefs with a plain `value` or a `getValue`
@@ -267,7 +267,7 @@
         "CA": ["en-CA", "en-GB", "en-US", "en-ZA"],
         "DE": ["de", "de-DE", "de-AT", "de-CH"]
       })[geo];
-      return !!locales && locales.includes(locale);
+      return false;
     }
   },
   {
