--- a/browser/components/newtab/lib/AboutPreferences.jsm.orig     2021-03-18 13:48:28.000000000 +0000
+++ b/browser/components/newtab/lib/AboutPreferences.jsm  2021-04-18 02:13:56.605573190 +0000
@@ -156,7 +156,6 @@
     this.renderPreferences(window, [
       ...PREFS_BEFORE_SECTIONS(featureConfig),
       ...sections,
-      ...PREFS_AFTER_SECTIONS(featureConfig),
     ]);
   }
 
@@ -216,6 +215,11 @@
       const { feed: name, titleString = {}, descString, nestedPrefs = [] } =
         prefData || {};
 
+      // Remove topstories section entirely
+      if (id === "topstories") {
+        return;
+      }
+
       // Don't show any sections that we don't want to expose in preferences UI
       if (shouldHidePref) {
         return;
@@ -244,19 +248,6 @@
 
       linkPref(checkbox, name, "bool");
 
-      // Specially add a link for stories
-      if (id === "topstories") {
-        const sponsoredHbox = createAppend("hbox", sectionVbox);
-        sponsoredHbox.setAttribute("align", "center");
-        sponsoredHbox.appendChild(checkbox);
-        checkbox.classList.add("tail-with-learn-more");
-
-        const link = createAppend("label", sponsoredHbox, { is: "text-link" });
-        link.classList.add("learn-sponsored");
-        link.setAttribute("href", sectionData.pref.learnMore.link.href);
-        document.l10n.setAttributes(link, sectionData.pref.learnMore.link.id);
-      }
-
       // Add more details for the section (e.g., description, more prefs)
       const detailVbox = createAppend("vbox", sectionVbox);
       detailVbox.classList.add("indent");
--- a/browser/components/newtab/lib/SectionsManager.jsm.orig      2020-07-29 22:35:14.254579199 +0000
+++ b/browser/components/newtab/lib/SectionsManager.jsm   2020-07-29 22:35:23.767082702 +0000
@@ -108,14 +108,6 @@
           name: "section.highlights.includeDownloads",
           titleString: "home-prefs-highlights-option-most-recent-download",
         },
-        {
-          name: "section.highlights.includePocket",
-          titleString: "home-prefs-highlights-option-saved-to-pocket",
-          hidden: !Services.prefs.getBoolPref(
-            "extensions.pocket.enabled",
-            true
-          ),
-        },
       ],
     },
     shouldHidePref: false,
--- a/browser/components/newtab/lib/DiscoveryStreamFeed.jsm.orig	2021-10-28 18:03:03.000000000 +0000
+++ b/browser/components/newtab/lib/DiscoveryStreamFeed.jsm	2021-11-06 03:00:17.374753813 +0000
@@ -1884,7 +1884,7 @@
 }) => ({
   lastUpdate: Date.now(),
   spocs: {
-    url: "https://spocs.getpocket.com/spocs",
+    url: "",
   },
   layout: [
     {
@@ -1971,7 +1971,7 @@
           feed: {
             embed_reference: null,
             url:
-              "https://getpocket.cdn.mozilla.net/v3/firefox/global-recs?version=3&consumer_key=$apiKey&locale_lang=$locale&region=$region&count=30",
+              "",
           },
           spocs: {
             probability: 1,
@@ -1986,47 +1986,8 @@
           properties: {
             alignment: "left-align",
             links: [
-              {
-                name: "Self improvement",
-                url:
-                  "https://getpocket.com/explore/self-improvement?utm_source=pocket-newtab",
-              },
-              {
-                name: "Food",
-                url:
-                  "https://getpocket.com/explore/food?utm_source=pocket-newtab",
-              },
-              {
-                name: "Entertainment",
-                url:
-                  "https://getpocket.com/explore/entertainment?utm_source=pocket-newtab",
-              },
-              {
-                name: "Health & fitness",
-                url:
-                  "https://getpocket.com/explore/health?utm_source=pocket-newtab",
-              },
-              {
-                name: "Science",
-                url:
-                  "https://getpocket.com/explore/science?utm_source=pocket-newtab",
-              },
-              {
-                name: "More recommendations ›",
-                url: "https://getpocket.com/explore?utm_source=pocket-newtab",
-              },
             ],
             extraLinks: [
-              {
-                name: "Career",
-                url:
-                  "https://getpocket.com/explore/career?utm_source=pocket-newtab",
-              },
-              {
-                name: "Technology",
-                url:
-                  "https://getpocket.com/explore/technology?utm_source=pocket-newtab",
-              },
             ],
             privacyNoticeURL: {
               url:
