diff -u browser/components/moz.build.orig browser/components/moz.build
--- a/browser/components/moz.build.orig   2020-12-16 16:42:10.547775427 +0000
+++ b/browser/components/moz.build        2020-12-16 17:03:30.568971423 +0000
@@ -43,7 +43,6 @@
     "originattributes",
     "ion",
     "places",
-    "pocket",
     "preferences",
     "privatebrowsing",
     "prompts",
diff -u browser/components/BrowserGlue.jsm.orig browser/components/BrowserGlue.jsm
--- a/browser/components/BrowserGlue.jsm.orig     2021-01-18 18:29:32.000000000 +0000
+++ b/browser/components/BrowserGlue.jsm  2021-02-03 00:47:10.357303253 +0000
@@ -73,7 +73,6 @@
   RFPHelper: "resource://gre/modules/RFPHelper.jsm",
   SafeBrowsing: "resource://gre/modules/SafeBrowsing.jsm",
   Sanitizer: "resource:///modules/Sanitizer.jsm",
-  SaveToPocket: "chrome://pocket/content/SaveToPocket.jsm",
   SearchSERPTelemetry: "resource:///modules/SearchSERPTelemetry.jsm",
   SessionStartup: "resource:///modules/sessionstore/SessionStartup.jsm",
   SessionStore: "resource:///modules/sessionstore/SessionStore.jsm",
@@ -1361,8 +1360,6 @@
       Normandy.init();
     }
 
-    SaveToPocket.init();
-
     AboutHomeStartupCache.init();
 
     Services.obs.notifyObservers(null, "browser-ui-startup-complete");
diff -r 164a57c0cdf0 browser/base/content/browser.xhtml
--- a/browser/base/content/browser.xhtml.orig     2021-04-18 02:08:26.551737670 +0000
+++ b/browser/base/content/browser.xhtml  2021-04-18 02:08:45.392755610 +0000
@@ -2007,12 +2007,6 @@
                        data-l10n-id="urlbar-page-action-button"
                        onmousedown="BrowserPageActions.mainButtonClicked(event);"
                        onkeypress="BrowserPageActions.mainButtonClicked(event);"/>
-                <image id="pocket-button"
-                       class="urlbar-icon urlbar-page-action"
-                       data-l10n-id="urlbar-pocket-button"
-                       role="button"
-                       hidden="true"
-                       onclick="BrowserPageActions.doCommandForAction(PageActions.actionForID('pocket'), event, this);"/>
                 <hbox id="star-button-box"
                       hidden="true"
                       class="urlbar-icon-wrapper urlbar-page-action"
diff -u browser/base/content/browser-context.inc.orig browser/base/content/browser-context.inc
--- a/browser/base/content/browser-context.inc.orig       2020-02-11 05:01:03.762178005 +0000
+++ b/browser/base/content/browser-context.inc    2020-02-11 05:18:24.730521197 +0000
@@ -76,9 +76,6 @@
       <menuitem id="context-savelink"
                 data-l10n-id="main-context-menu-save-link"
                 oncommand="gContextMenu.saveLink();"/>
-      <menuitem id="context-savelinktopocket"
-                data-l10n-id="main-context-menu-save-link-to-pocket"
-                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>
       <menuitem id="context-copyemail"
                 data-l10n-id="main-context-menu-copy-email"
                 oncommand="gContextMenu.copyEmail();"/>
@@ -216,9 +213,6 @@
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
                 oncommand="gContextMenu.savePageAs();"/>
-      <menuitem id="context-pocket"
-                data-l10n-id="main-context-menu-save-to-pocket"
-                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
       <menuseparator id="context-sep-sendpagetodevice" class="sync-ui-item"
                      hidden="true"/>
       <menu id="context-sendpagetodevice"
