diff -r 84219fbf133c browser/extensions/moz.build
--- a/browser/extensions/moz.build	Wed Oct 17 18:50:12 2018 +0000
+++ b/browser/extensions/moz.build	Tue Oct 23 16:18:37 2018 -0700
@@ -5,11 +5,9 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DIRS += [
-    'aushelper',
     'formautofill',
     'onboarding',
     'pdfjs',
-    'pocket',
     'screenshots',
     'webcompat',
     'webcompat-reporter'
