diff -r 8337ebb86a42 browser/components/moz.build
--- a/browser/components/moz.build	Thu Dec 06 15:18:54 2018 -0500
+++ b/browser/components/moz.build	Tue Dec 11 20:42:07 2018 -0800
@@ -44,7 +44,6 @@
     'newtab',
     'originattributes',
     'places',
-    'pocket',
     'preferences',
     'privatebrowsing',
     'resistfingerprinting',
diff -r 164a57c0cdf0 browser/components/nsBrowserGlue.js
--- a/browser/components/nsBrowserGlue.js	Thu Mar 14 19:46:30 2019 +0200
+++ b/browser/components/nsBrowserGlue.js	Thu Mar 21 16:29:36 2019 -0700
@@ -426,7 +426,6 @@
   RemoteSettings: "resource://services-settings/remote-settings.js",
   SafeBrowsing: "resource://gre/modules/SafeBrowsing.jsm",
   Sanitizer: "resource:///modules/Sanitizer.jsm",
-  SaveToPocket: "chrome://pocket/content/SaveToPocket.jsm",
   SearchTelemetry: "resource:///modules/SearchTelemetry.jsm",
   SessionStartup: "resource:///modules/sessionstore/SessionStartup.jsm",
   SessionStore: "resource:///modules/sessionstore/SessionStore.jsm",
@@ -1081,7 +1080,6 @@
 
     Normandy.init();
 
-    SaveToPocket.init();
     Services.obs.notifyObservers(null, "browser-ui-startup-complete");
   },
 
diff -r 164a57c0cdf0 browser/base/content/browser.xul
--- a/browser/base/content/browser.xul	Thu Mar 14 19:46:30 2019 +0200
+++ b/browser/base/content/browser.xul	Thu Mar 21 16:31:48 2019 -0700
@@ -573,7 +573,6 @@
            position="bottomcenter topright"
            tabspecific="true"
            noautofocus="true"
-           pocket-title="&saveToPocketCmd.label;"
            copyURL-title="&pageAction.copyLink.label;"
            emailLink-title="&emailPageCmd.label;"
            sendToDevice-notReadyTitle="&sendToDevice.syncNotReady.label;"
@@ -1036,20 +1035,6 @@
                        tooltiptext="&pageActionButton.tooltip;"
                        onmousedown="BrowserPageActions.mainButtonClicked(event);"
                        onkeypress="BrowserPageActions.mainButtonClicked(event);"/>
-                <hbox id="pocket-button-box"
-                      hidden="true"
-                      class="urlbar-icon-wrapper urlbar-page-action"
-                      onclick="BrowserPageActions.doCommandForAction(PageActions.actionForID('pocket'), event, this);">
-                  <image id="pocket-button"
-                         class="urlbar-icon"
-                         tooltiptext="&pocketButton.tooltiptext;"
-                         role="button"/>
-                  <hbox id="pocket-button-animatable-box">
-                    <image id="pocket-button-animatable-image"
-                           tooltiptext="&pocketButton.tooltiptext;"
-                           role="presentation"/>
-                  </hbox>
-                </hbox>
                 <hbox id="star-button-box"
                       hidden="true"
                       class="urlbar-icon-wrapper urlbar-page-action"
diff -r 164a57c0cdf0 browser/base/content/browser-context.inc
--- a/browser/base/content/browser-context.inc	Thu Mar 14 19:46:30 2019 +0200
+++ b/browser/base/content/browser-context.inc	Thu Mar 21 16:32:53 2019 -0700
@@ -90,10 +90,6 @@
                 label="&saveLinkCmd.label;"
                 accesskey="&saveLinkCmd.accesskey;"
                 oncommand="gContextMenu.saveLink();"/>
-      <menuitem id="context-savelinktopocket"
-                label="&saveLinkToPocketCmd.label;"
-                accesskey="&saveLinkToPocketCmd.accesskey;"
-                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>
       <menuitem id="context-copyemail"
                 label="&copyEmailCmd.label;"
                 accesskey="&copyEmailCmd.accesskey;"
@@ -263,10 +259,6 @@
                 label="&savePageCmd.label;"
                 accesskey="&savePageCmd.accesskey2;"
                 oncommand="gContextMenu.savePageAs();"/>
-      <menuitem id="context-pocket"
-                label="&saveToPocketCmd.label;"
-                accesskey="&saveToPocketCmd.accesskey;"
-                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
       <menuseparator id="context-sep-sendpagetodevice" class="sync-ui-item"
                      hidden="true"/>
       <menu id="context-sendpagetodevice"
