diff -u browser/components/newtab/lib/ActivityStream.jsm.orig browser/components/newtab/lib/ActivityStream.jsm
--- a/browser/components/newtab/lib/ActivityStream.jsm.orig       2021-04-15 23:11:47.000000000 +0000
+++ b/browser/components/newtab/lib/ActivityStream.jsm    2021-04-20 00:06:32.633397507 +0000
@@ -117,10 +117,7 @@
 
 // Determine if spocs should be shown for a geo/locale
 function showSpocs({ geo }) {
-  const spocsGeoString =
-    Services.prefs.getStringPref(REGION_SPOCS_CONFIG) || "";
-  const spocsGeo = spocsGeoString.split(",").map(s => s.trim());
-  return spocsGeo.includes(geo);
+  return false;
 }
 
 // Configure default Activity Stream prefs with a plain `value` or a `getValue`
@@ -147,15 +144,10 @@
           hidden: !PREFS_CONFIG.get("feeds.system.topstories").getValue(args),
           provider_icon: "pocket",
           provider_name: "Pocket",
-          read_more_endpoint:
-            "https://getpocket.com/explore/trending?src=fx_new_tab",
-          stories_endpoint: `https://getpocket.cdn.mozilla.net/v3/firefox/global-recs?version=3&consumer_key=$apiKey&locale_lang=${
-            args.locale
-          }&feed_variant=${
-            showSpocs(args) ? "default_spocs_on" : "default_spocs_off"
-          }`,
-          stories_referrer: "https://getpocket.com/recommendations",
-          topics_endpoint: `https://getpocket.cdn.mozilla.net/v3/firefox/trending-topics?version=2&consumer_key=$apiKey&locale_lang=${args.locale}`,
+          read_more_endpoint: "",
+          stories_endpoint: "",
+          stories_referrer: "",
+          topics_endpoint: "",
           show_spocs: showSpocs(args),
         }),
     },
@@ -180,7 +172,7 @@
     {
       title:
         "Show sponsored cards in spoc experiment (show_spocs in topstories.options has to be set to true as well)",
-      value: true,
+      value: false,
     },
   ],
   [
@@ -220,21 +212,21 @@
     "feeds.snippets",
     {
       title: "Show snippets on activity stream",
-      value: true,
+      value: false,
     },
   ],
   [
     "topSitesRows",
     {
       title: "Number of rows of Top Sites to display",
-      value: 1,
+      value: 2,
     },
   ],
   [
     "telemetry",
     {
       title: "Enable system error and usage data collection",
-      value: true,
+      value: false,
       value_local_dev: false,
     },
   ],
@@ -242,7 +234,7 @@
     "telemetry.ut.events",
     {
       title: "Enable Unified Telemetry event data collection",
-      value: AppConstants.EARLY_BETA_OR_EARLIER,
+      value: false,
       value_local_dev: false,
     },
   ],
@@ -250,7 +242,7 @@
     "telemetry.structuredIngestion.endpoint",
     {
       title: "Structured Ingestion telemetry server endpoint",
-      value: "https://incoming.telemetry.mozilla.org/submit",
+      value: "",
     },
   ],
   [
@@ -274,7 +266,7 @@
     {
       title:
         "Boolean flag that decides whether or not to show saved Pocket stories in highlights.",
-      value: true,
+      value: false,
     },
   ],
   [
@@ -303,7 +295,7 @@
     "sectionOrder",
     {
       title: "The rendering order for the sections",
-      value: "topsites,topstories,highlights",
+      value: "topsites,highlights",
     },
   ],
   [
@@ -320,21 +312,7 @@
         "An ordered, comma-delimited list of search shortcuts that we should try and pin",
       // This pref is dynamic as the shortcuts vary depending on the region
       getValue: ({ geo }) => {
-        if (!geo) {
-          return "";
-        }
-        const searchShortcuts = [];
-        if (geo === "CN") {
-          searchShortcuts.push("baidu");
-        } else if (["BY", "KZ", "RU", "TR"].includes(geo)) {
-          searchShortcuts.push("yandex");
-        } else {
-          searchShortcuts.push("google");
-        }
-        if (["DE", "FR", "GB", "IT", "JP", "US"].includes(geo)) {
-          searchShortcuts.push("amazon");
-        }
-        return searchShortcuts.join(",");
+        return "";
       },
     },
   ],
@@ -397,13 +375,12 @@
         return JSON.stringify({
           api_key_pref: "extensions.pocket.oAuthConsumerKey",
           collapsible: true,
-          enabled: true,
+          enabled: false,
           show_spocs: showSpocs({ geo }),
           hardcoded_layout: true,
           personalized: true,
           // This is currently an exmple layout used for dev purposes.
-          layout_endpoint:
-            "https://getpocket.cdn.mozilla.net/v3/newtab/layout?version=1&consumer_key=$apiKey&layout_variant=basic",
+          layout_endpoint: "",
         });
       },
     },
@@ -413,7 +390,7 @@
     {
       title:
         "Endpoint prefixes (comma-separated) that are allowed to be requested",
-      value: "https://getpocket.cdn.mozilla.net/,https://spocs.getpocket.com/",
+      value: "",
     },
   ],
   [
@@ -466,7 +443,7 @@
     {
       title:
         "Endpoint for when a user opts-out of sponsored content to delete the user's data from the ad server.",
-      value: "https://spocs.getpocket.com/user",
+      value: "",
     },
   ],
   [
@@ -525,49 +502,7 @@
       "System pref that fetches content recommendations from a configurable content provider",
     // Dynamically determine if Pocket should be shown for a geo / locale
     getValue: ({ geo, locale }) => {
-      // If we don't have geo, we don't want to flash the screen with stories while geo loads.
-      // Best to display nothing until geo is ready.
-      if (!geo) {
-        return false;
-      }
-      const preffedRegionsBlockString =
-        Services.prefs.getStringPref(REGION_STORIES_BLOCK) || "";
-      const preffedRegionsString =
-        Services.prefs.getStringPref(REGION_STORIES_CONFIG) || "";
-      const preffedLocaleListString =
-        Services.prefs.getStringPref(LOCALE_LIST_CONFIG) || "";
-      const preffedBlockRegions = preffedRegionsBlockString
-        .split(",")
-        .map(s => s.trim());
-      const preffedRegions = preffedRegionsString.split(",").map(s => s.trim());
-      const preffedLocales = preffedLocaleListString
-        .split(",")
-        .map(s => s.trim());
-      const locales = {
-        US: ["en-CA", "en-GB", "en-US"],
-        CA: ["en-CA", "en-GB", "en-US"],
-        GB: ["en-CA", "en-GB", "en-US"],
-        AU: ["en-CA", "en-GB", "en-US"],
-        NZ: ["en-CA", "en-GB", "en-US"],
-        IN: ["en-CA", "en-GB", "en-US"],
-        IE: ["en-CA", "en-GB", "en-US"],
-        ZA: ["en-CA", "en-GB", "en-US"],
-        CH: ["de"],
-        BE: ["de"],
-        DE: ["de"],
-        AT: ["de"],
-        IT: ["it"],
-        FR: ["fr"],
-        ES: ["es"],
-        PL: ["pl"],
-        JP: ["ja", "ja-JP-mac"],
-      }[geo];
-
-      const regionBlocked = preffedBlockRegions.includes(geo);
-      const localeEnabled = locale && preffedLocales.includes(locale);
-      const regionEnabled =
-        preffedRegions.includes(geo) && !!locales && locales.includes(locale);
-      return !regionBlocked && (localeEnabled || regionEnabled);
+      return false;
     },
   },
   {
@@ -580,7 +515,7 @@
     name: "telemetry",
     factory: () => new TelemetryFeed(),
     title: "Relays telemetry-related actions to PingCentre",
-    value: true,
+    value: false,
   },
   {
     name: "favicon",
@@ -598,13 +533,13 @@
     name: "recommendationproviderswitcher",
     factory: () => new RecommendationProviderSwitcher(),
     title: "Handles switching between two types of personality providers",
-    value: true,
+    value: false,
   },
   {
     name: "discoverystreamfeed",
     factory: () => new DiscoveryStreamFeed(),
     title: "Handles new pocket ui for the new tab page",
-    value: true,
+    value: false,
   },
 ];
 
diff -u browser/components/newtab/lib/DiscoveryStreamFeed.jsm.orig browser/components/newtab/lib/DiscoveryStreamFeed.jsm
--- a/browser/components/newtab/lib/DiscoveryStreamFeed.jsm.orig  2020-12-12 00:35:33.000000000 +0000
+++ b/browser/components/newtab/lib/DiscoveryStreamFeed.jsm       2020-12-16 17:10:57.009219456 +0000
@@ -1943,7 +1943,7 @@
 getHardcodedLayout = isBasicLayout => ({
   lastUpdate: Date.now(),
   spocs: {
-    url: "https://spocs.getpocket.com/spocs",
+    url: "",
   },
   layout: [
     {
@@ -2023,7 +2023,7 @@
           feed: {
             embed_reference: null,
             url:
-              "https://getpocket.cdn.mozilla.net/v3/firefox/global-recs?version=3&consumer_key=$apiKey&locale_lang=$locale&region=$region&count=30",
+              "",
           },
           spocs: {
             probability: 1,
@@ -2048,35 +2048,6 @@
           properties: {
             alignment: "left-align",
             links: [
-              {
-                name: "Self Improvement",
-                url:
-                  "https://getpocket.com/explore/self-improvement?utm_source=pocket-newtab",
-              },
-              {
-                name: "Food",
-                url:
-                  "https://getpocket.com/explore/food?utm_source=pocket-newtab",
-              },
-              {
-                name: "Entertainment",
-                url:
-                  "https://getpocket.com/explore/entertainment?utm_source=pocket-newtab",
-              },
-              {
-                name: "Health",
-                url:
-                  "https://getpocket.com/explore/health?utm_source=pocket-newtab",
-              },
-              {
-                name: "Science",
-                url:
-                  "https://getpocket.com/explore/science?utm_source=pocket-newtab",
-              },
-              {
-                name: "More Recommendations ›",
-                url: "https://getpocket.com/explore?utm_source=pocket-newtab",
-              },
             ],
           },
           header: {
--- a/browser/components/newtab/lib/AboutPreferences.jsm.orig     2021-03-18 13:48:28.000000000 +0000
+++ b/browser/components/newtab/lib/AboutPreferences.jsm  2021-04-18 02:13:56.605573190 +0000
@@ -156,7 +156,6 @@
     this.renderPreferences(window, [
       ...PREFS_BEFORE_SECTIONS(featureConfig),
       ...sections,
-      ...PREFS_AFTER_SECTIONS(featureConfig),
     ]);
   }
 
@@ -216,6 +215,11 @@
       const { feed: name, titleString = {}, descString, nestedPrefs = [] } =
         prefData || {};
 
+      // Remove topstories section entirely
+      if (id === "topstories") {
+        return;
+      }
+
       // Don't show any sections that we don't want to expose in preferences UI
       if (shouldHidePref) {
         return;
@@ -244,19 +248,6 @@
 
       linkPref(checkbox, name, "bool");
 
-      // Specially add a link for stories
-      if (id === "topstories") {
-        const sponsoredHbox = createAppend("hbox", sectionVbox);
-        sponsoredHbox.setAttribute("align", "center");
-        sponsoredHbox.appendChild(checkbox);
-        checkbox.classList.add("tail-with-learn-more");
-
-        const link = createAppend("label", sponsoredHbox, { is: "text-link" });
-        link.classList.add("learn-sponsored");
-        link.setAttribute("href", sectionData.pref.learnMore.link.href);
-        document.l10n.setAttributes(link, sectionData.pref.learnMore.link.id);
-      }
-
       // Add more details for the section (e.g., description, more prefs)
       const detailVbox = createAppend("vbox", sectionVbox);
       detailVbox.classList.add("indent");
--- a/browser/components/newtab/lib/SectionsManager.jsm.orig      2020-07-29 22:35:14.254579199 +0000
+++ b/browser/components/newtab/lib/SectionsManager.jsm   2020-07-29 22:35:23.767082702 +0000
@@ -108,14 +108,6 @@
           name: "section.highlights.includeDownloads",
           titleString: "home-prefs-highlights-option-most-recent-download",
         },
-        {
-          name: "section.highlights.includePocket",
-          titleString: "home-prefs-highlights-option-saved-to-pocket",
-          hidden: !Services.prefs.getBoolPref(
-            "extensions.pocket.enabled",
-            true
-          ),
-        },
       ],
     },
     shouldHidePref: false,
