--- a/python/mozbuild/mozbuild/controller/building.py.old	2020-04-12 22:07:22.126080489 +0000
+++ b/python/mozbuild/mozbuild/controller/building.py	2020-04-12 22:08:16.709010310 +0000
@@ -586,6 +586,10 @@
             if self.footer:
                 self.footer.clear()
 
+            #import string
+            #msg = filter(lambda x: x in set(string.printable), msg)
+            msg = msg.encode('ascii', errors='ignore')
+
             self.fh.write(msg)
             self.fh.write('\n')
 
